modules/hostdb/hostentry.go

```diff
@@ -0,0 +1,92 @@
+package hostdb
+
+import (
+	"math/big"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+var (
+	// Because most weights would otherwise be fractional, we set the base
+	// weight to 10^80 to give ourselves lots of precision when determing the
+	// weight of an entry.
+	baseWeight = types.NewCurrency(new(big.Int).Exp(big.NewInt(10), big.NewInt(80), nil))
+)
+
+// threadedProbeHost tries to fetch the settings of a host. If successful, the
+// host is put in the set of active hosts. If unsuccessful, the host id deleted
+// from the set of active hosts.
+func (hdb *HostDB) threadedProbeHost(entry *modules.HostEntry) {
+	// If the host is already in the set of active hosts, remove it. It will be
+	// re-added if it responds to the probing, and left out if not. The db
+	// needs to be locked for this operation, but then unlocked while the
+	// network communication is happening.
+	id := hdb.mu.Lock()
+	{
+		node, exists := hdb.activeHosts[entry.IPAddress]
+		if exists {
```

Comment:
I felt like I needed to highlight the Lock and Unlock, but that separating them with newlines didn't make it clear enough. So I put the locked code a bare block.